dask/dataframe/io/parquet.py

```diff
@@ -268,50 +334,150 @@ def _read_fastparquet(fs, fs_token, paths, columns=None, filters=None,
     if out_type == Series:
         assert len(meta.columns) == 1
         meta = meta[meta.columns[0]]
-
-    name = 'read-parquet-' + tokenize(fs_token, paths, all_columns, filters,
+    return (meta, filters, index_names, out_type, all_columns, index_names,
+            storage_name_mapping)
+
+
+def _read_fp_multifile(fs, fs_token, paths, columns=None,
+                       categories=None, index=None):
+    """Read dataset with fastparquet by assuming metadata from first file"""
+    from fastparquet import ParquetFile
+    from fastparquet.util import analyse_paths, get_file_scheme
```

Comment:
These are always available on the oldest fast parquet we support?

Reply:
Both been around over a year